src/components/hooks/usePreviousActivities.js

```diff
@@ -0,0 +1,67 @@
+import {useContext, useEffect, useRef, useState} from 'react';
+import {Observable} from 'rxjs';
+import {flatMap} from 'rxjs/operators';
+
+import {AdapterContext} from '../../components/';
+
+/**
+ * Callback function to execute once data has been fetched.
+ *
+ * @callback previousActivitiesCallback
+ * @param {Array.<string|ActivityDate>} previousActivities Array of previous activities fetched
+ * @returns undefined
+ */
+
+/**
+ * Custom hook that returns a reference to the DOM element of the last activity.
+ * Requests previous activity data for the given room, and executes
+ * the given callback with that data.
+ * It completes once there is enough data to fill the viewport of the
+ * parent container, finishing by scrolling into view the last activity.
+ *
+ * @param {string} roomID  ID of the room for which to load data.
+ * @param {object} elementRef  DOM reference of the parent component.
+ * @param {previousActivitiesCallback} callback  Callback to execute once data has been fetched.
+ * @returns {object}  DOM reference to the the last activity.
+ */
+export default function usePreviousActivities(roomID, elementRef, callback) {
```

Comment:
If this hook wasn't defined, wouldn't the top scroll hook catch this case because the scroll top would be the viewport top?

Reply:
It would not, there are 2 different but related use cases.
`useActivityScroll` listens to scroll events, and while there is more data to load, it will load more of the previous activities as you're scrolling up.
`usePreviousActivities` kinda works as an initializer, which loads as much data as necessary to try to overflow the container - based on its size - and then leaves you on the most recent activity. Without `usePreviousActivities`, `useActivityScroll` wouldn't work, because if there is no content loaded or not enough content loaded, the activity stream container will not overflow, and therefore won't have a scroll bar to trigger the scroll event (load more -even older - data). Now, if there is not enough previous activities to overflow the container to begin with, there's no need to listen for scroll events because, well, there's nothing more to load on scroll up.
Before i had all this in `useActivityStream` (combined with the subscription to future events), but i thought separating them out would be more maintainable.

Reply:
I like `useOverflowActivities` a lot. It is less confusing to me since `usePreviousActivities` sounds like it is a hook specifically for `getPreviousRoomActivities`.

Reply:
Renamed to `useOverflowActivities`.